### ISSUE TITLE
run_algorithm(): local variable 'data' referenced before assignment

### ISSUE THREAD

**Issue by jkleint** [quantopian/zipline#1506]
I've just installed zipline, done a `zipline ingest`, and would like to run a backtest using the API.  But `run_algorithm()` gives an `UnboundLocalError` right out of the box:

```
#!/usr/bin/env python3
from datetime import datetime

import zipline

def initialize(context):
    pass
    
def handle_data(context, data):
    pass        
   
if __name__ == "__main__":
    zipline.run_algorithm(initialize=initialize, handle_data=handle_data, capital_base=1000, start=datetime(2016, 1, 1), end=datetime(2016, 9, 1))
```

```
$ ./databug.py 

quantopian/python/lib/python3.5/site-packages/pandas/tseries/index.py:814: PerformanceWarning: Non-vectorized DateOffset being applied to Series or DatetimeIndex
  "or DatetimeIndex", PerformanceWarning)
quantopian/python/lib/python3.5/site-packages/pandas/tseries/index.py:814: PerformanceWarning: Non-vectorized DateOffset being applied to Series or DatetimeIndex
  "or DatetimeIndex", PerformanceWarning)
quantopian/python/lib/python3.5/site-packages/pandas/tseries/index.py:814: PerformanceWarning: Non-vectorized DateOffset being applied to Series or DatetimeIndex
  "or DatetimeIndex", PerformanceWarning)
Traceback (most recent call last):
  File "./databug.py", line 13, in <module>
    zipline.run_algorithm(initialize=initialize, handle_data=handle_data, capital_base=1000, start=datetime(2016, 1, 1), end=datetime(2016, 9, 1))
  File "quantopian/python/lib/python3.5/site-packages/zipline/utils/run_algo.py", line 361, in run_algorithm
    environ=environ,
  File "quantopian/python/lib/python3.5/site-packages/zipline/utils/run_algo.py", line 132, in _run
    env = TradingEnvironment(asset_db_path=connstr)
  File "quantopian/python/lib/python3.5/site-packages/zipline/finance/trading.py", line 96, in __init__
    self.bm_symbol,
  File "quantopian/python/lib/python3.5/site-packages/zipline/data/loader.py", line 164, in load_market_data
    trading_day,
  File "quantopian/python/lib/python3.5/site-packages/zipline/data/loader.py", line 252, in ensure_benchmark_data
    if not has_data_for_dates(data, first_date, last_date):
UnboundLocalError: local variable 'data' referenced before assignment
```

## Environment

* Operating System: `Linux x 4.4.0-36-generic #55-Ubuntu SMP Thu Aug 11 18:01:55 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux`
* Python Version: `Python 3.5.2`
* Python Bitness: `64`
* How did you install Zipline: `pip`
* Python packages: 
```
alembic==0.8.8
astroid==1.4.8
bcolz==0.12.1
Bottleneck==1.1.0
cachetools==1.1.6
click==6.6
contextlib2==0.5.4
cycler==0.10.0
cyordereddict==1.0.0
Cython==0.24.1
decorator==4.0.10
empyrical==0.2.0
funcsigs==1.0.2
intervaltree==2.1.0
ipython==5.1.0
ipython-genutils==0.1.0
isort==4.2.5
lazy-object-proxy==1.2.2
Logbook==1.0.0
lru-dict==1.1.4
Mako==1.0.4
MarkupSafe==0.23
matplotlib==1.5.3
mccabe==0.5.2
mock==2.0.0
multipledispatch==0.4.8
networkx==1.11
numexpr==2.6.1
numpy==1.11.1
pandas==0.18.1
pandas-datareader==0.2.1
patsy==0.4.1
pbr==1.10.0
pexpect==4.2.1
pickleshare==0.7.4
pkg-resources==0.0.0
prompt-toolkit==1.0.7
ptyprocess==0.5.1
pyfolio==0.5.1
Pygments==2.1.3
pylint==1.6.4
pyparsing==2.1.9
python-dateutil==2.5.3
python-editor==1.0.1
pytz==2016.6.1
requests==2.11.1
requests-file==1.4
scikit-learn==0.17.1
scipy==0.18.1
seaborn==0.7.1
simplegeneric==0.8.1
six==1.10.0
sortedcontainers==1.5.3
SQLAlchemy==1.0.15
statsmodels==0.6.1
Theano==0.8.2
toolz==0.8.0
traitlets==4.3.0
wcwidth==0.1.7
wrapt==1.10.8
zipline==1.0.2+103.g9d7049a
```

The same happens with zipline 1.0.2 from PyPI.

**Comment by jkleint** [comment]
One-line fix in #1513 to re-raise swallowed exceptions.

It was a network error, incidentally.

**Comment by ssanderson** [comment]
Fixed by https://github.com/quantopian/zipline/pull/1513.
